mesonbuild/coredata.py

```diff
@@ -782,7 +782,8 @@ def update_project_options(self, options: 'MutableKeyedOptionDictType') -> None:
                 try:
                     value.set_value(oldval.value)
                 except MesonException:
-                    mlog.warning(f'Old value(s) of {key} are no longer valid, resetting to default ({value.value}).')
+                    mlog.warning(f'Old value(s) of {key} are no longer valid, resetting to default ({value.value}).',
+                                 fatal=False)
```

Comment:
I'm conflicted about this. On the one hand, the user absolutely can and really should do something about this, unlike the MSVC std case where the user cannot fix Microsoft.

On the other hand, this may be triggered by `ninja` updating build.ninja by running a reconfigure, which drops --fatal-meson-warnings anyway.

On the other other hand, in that case the value of fatal= is irrelevant...

Do we really want to avoid failing here?